appcelerator/titanium_mobile
# fix(android): ListView's SearchBar/SearchView wrongly overlaps rows as of 8.0.1

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27094

**Summary:**
- Caused by us no longer auto-assigning IDs to all views in 8.0.1.
  * Old code's `RelativeLayout` was trying to layout the `ListView` beneath search view via ID.
  * We don't want to arbitrarily assign IDs to views anymore to avoid random crashes when Android OS "restores" an activity.
- Changed its view group from `RelativeLayout` to a vertical `LinearLayout`.

**Test:**
1. Build and run the below code on Android.
2. Verify that the `SearchBar` does not overlap the `ListView` rows.
3. In the code, comment out: `Ti.UI.createSearchBar({ showCancel: true })`
4. In the code, uncomment: `Ti.UI.createSearchBar({ showCancel: true, height: "120dp" })`
5. Build and run on Android.
6. Verify that the `SearchBar` is taller and does not overlap `ListView` rows.
7. Comment out: `Ti.UI.createSearchBar({ showCancel: true, height: "120dp" })`
8. Uncomment: `Ti.UI.Android.createSearchView()`
9. Build and run on Android.
10. Verify that the `SearchView` does not overlap the `ListView` rows.
11. Comment out: `Ti.UI.Android.createSearchView()`
12. Uncomment: `Ti.UI.Android.createSearchView({ height: "120dp" })`
13. Build and run on Android.
14. Verify that the `SearchView` is taller and does not overlap `ListView` rows.

```javascript
function createListViewSectionItems() {
	var items = [];
	for (var index = 1; index <= 100; index++) {
		items.push({ properties: { title: "Row " + index } });
	}
	return items;
}

var window = Ti.UI.createWindow();
window.add(Ti.UI.createListView({
	sections: [Ti.UI.createListSection({
		headerTitle: "ListView",
		items: createListViewSectionItems(),
	})],
	searchView: Ti.UI.createSearchBar({ showCancel: true }),
//	searchView: Ti.UI.createSearchBar({ showCancel: true, height: "120dp" }),
//	searchView: Ti.UI.Android.createSearchView(),
//	searchView: Ti.UI.Android.createSearchView({ height: "120dp" }),
}));
window.open();
```
